pkg/gateway/server/user.go

```diff
@@ -361,3 +386,13 @@ func (s *Server) restrictGroups(ctx context.Context, gptscriptClient *gptscript.
 
 	return groups, nil
 }
+
+func userIsBasicOrPower(u user.Info) bool {
+	for _, group := range u.GetGroups() {
+		switch group {
+		case types2.GroupPowerUserPlus, types2.GroupAuditor, types2.GroupAdmin, types2.GroupOwner:
```

Comment:
I'm considering Auditor as "higher" than PU (even though it is orthogonal) here, because it already has permissions to list all users.